src/services/etherspot.js

```diff
@@ -0,0 +1,75 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2019 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import {
+  Sdk as EtherspotSdk,
+  NetworkNames,
+  Account as EtherspotAccount,
+  Accounts as EtherspotAccounts,
+  EnvNames,
+} from 'etherspot';
+
+// utils
+import { reportErrorLog } from 'utils/common';
+
+// config
+import { getEnv } from 'configs/envConfig';
+
+
+class EtherspotService {
+  sdk: EtherspotSdk;
+  subscription: any;
+
+  async init(privateKey: string) {
+    const networkName = getEnv().NETWORK_PROVIDER === 'homestead'
+      ? NetworkNames.Mainnet
+      : NetworkNames.Kovan;
+
+    const envName = getEnv().NETWORK_PROVIDER === 'homestead'
+      ? EnvNames.MainNets
+      : EnvNames.TestNets;
+
+    this.sdk = new EtherspotSdk(privateKey, { env: envName, networkName });
+
+    await this.sdk.computeContractAccount({ sync: true }).catch((error) => {
+      reportErrorLog('EtherspotService init computeContractAccount failed', { error });
+    });
+  }
+
+  getAccounts(): Promise<?EtherspotAccount[]> {
+    return this.sdk.getConnectedAccounts()
+      .then(({ items }: EtherspotAccounts) => items)
+      .catch((error) => {
+        reportErrorLog('EtherspotService getAccounts -> getConnectedAccounts failed', { error });
+        return null;
+      });
```

Comment:
Is there any particular reason why you want it to be wrapped with `try...catch` exception handler? If it's because of SDK instance being empty and therefore failing on selected method call then we can just add a check before, wdyt?

Reply:
This should be exactly the same code as replaced one, just coded as `async/await` instead of `.then().catch()`. This is a bit of nitpicking form my side :-) but I thought that as you used `async/await` in other methods from the same file, it would make it more consistent.

Reply:
This is single method that has been copied over, but there are many more in the full implementation branch.

Actually the reason of choosing `async/await` over `try/catch` at this piece is just that it's easier to destruct the response object with less code.

I think let's keep it as it is if that won't bother you too much 😂